src/components/NcAppNavigation/NcAppNavigation.vue

```diff
@@ -134,10 +134,128 @@ emit('toggle-navigation', {
 
 </docs>
 
+<script setup lang="ts">
+import type { FocusTrap } from 'focus-trap'
+
+import { emit, subscribe, unsubscribe } from '@nextcloud/event-bus'
+import { createFocusTrap } from 'focus-trap'
+import { inject, onMounted, onUnmounted, ref, useTemplateRef, warn, watch } from 'vue'
+import NcAppNavigationList from '../NcAppNavigationList/index.js'
+import NcAppNavigationToggle from './NcAppNavigationToggle.vue'
+import { useIsMobile } from '../../composables/useIsMobile/index.ts'
+import { getTrapStack } from '../../utils/focusTrap.ts'
+
+defineProps<{
+	/**
+	 * The aria label to describe the navigation
+	 */
+	ariaLabel?: string
+
+	/**
+	 * aria-labelledby attribute to describe the navigation
+	 */
+	ariaLabelledby?: string
+}>()
+
+const setHasAppNavigation = inject<(v: boolean) => void>('NcContent:setHasAppNavigation', () => warn('NcAppNavigation is not mounted inside NcContent, this is probably an error.'), false)
+
+const appNavigationContainer = useTemplateRef('appNavigationContainer')
+const focusTrap = ref<FocusTrap>()
```

Comment:
It is a general variable, not a reactive state

```suggestion
let focusTrap: FocusTrap
```

Reply:
Or, if we really want a reactive variable just in case, let's not make entire trap object deep reactive.
```suggestion
const focusTrap = shalowRef<FocusTrap>()
```

Reply:
TBH I did not wanted to touch implementation - it was in `data` thus reactive.
But I guess it does not need to be reactive here.

Reply:
But I appreciate single-change refactoring